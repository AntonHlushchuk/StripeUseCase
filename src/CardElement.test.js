import React from 'react';
import renderer from 'react-test-renderer';
import CardComponent from './CardElement';  // adjust path to where your component is

jest.mock('@stripe/react-stripe-js', () => ({
    CardElement: () => <div data-testid="CardElement">CardElement</div>,
    useStripe: () => null,
    useElements: () => null,
}));

describe('CardComponent', () => {
    it('renders correctly', () => {
        const tree = renderer.create(<CardComponent />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
