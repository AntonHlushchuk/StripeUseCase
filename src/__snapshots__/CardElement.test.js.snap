// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardComponent renders correctly 1`] = `
<form
  onSubmit={[Function]}
  style={
    Object {
      "alignItems": "center",
      "border": "1px solid #ccc",
      "display": "flex",
      "flexDirection": "column",
      "height": "100vh",
      "justifyContent": "center",
      "padding": "20px",
      "width": "100%",
    }
  }
>
  <div
    style={
      Object {
        "height": "50px",
        "marginBottom": "20px",
        "width": "300px",
      }
    }
  >
    <div
      data-testid="CardElement"
    >
      CardElement
    </div>
  </div>
  <button
    disabled={true}
    type="submit"
  >
    Pay
  </button>
</form>
`;
